tests/Doctrine/Tests/DBAL/Functional/ResultCacheTest.php

```diff
@@ -212,7 +213,10 @@ public function testDontFinishNoCache(): void
         self::assertCount(2, $this->sqlLogger->queries);
     }
 
-    public function testFetchAllSavesCache(): void
+    /**
+     * @dataProvider fetchAllProvider
```

Comment:
This is no longer solely about `fetchAll()`

```suggestion
     * @dataProvider cacheSavingMethodProvider
```

Reply:
All methods being tested fetch all records, that's why they save cache, they are not the methods saving cache. From the fetching a single `fetchAll()` method, the test transitioned to testing all methods fetching all records.

I'd just rename the test and leave everything as is.